### PR TITLE
Make postcode validation more robust

### PR DESCRIPTION
**Pull request by urfolomeus**
There were two issues that we were dealing with here:

1. Entering a postcode without a space between parts fails as invalid.
2. Postcodes with a valid shape but invalid code where being accepted (i.e. 1234 567).

This PR introduces a better postcode validator. It also allows users to add a valid postcode without spaces and will save it with a space in the right place.

demo

### CODE REVIEW

File: app/models/office.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require 'uk_postcode'
```
Comment (munckymagik) [inline app/models/office.rb]: Do we need to require it explicitly?

Comment (urfolomeus) [inline app/models/office.rb]: Yes, this is not Rails :)

Comment (munckymagik) [inline app/models/office.rb]: It's not Rails?

Comment (urfolomeus) [inline app/models/office.rb]: AFAIK there is no way to auto-require from a gemspec?

Comment (urfolomeus) [inline app/models/office.rb]: It doesn't work if you don't require the gem, so unless there's a way to auto-require that I don't know about ...

File: spec/support/shared_examples/geocodable_examples.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
   def modify_address(subject)
     subject.send("#{address_field_name}=", address_field_updated_value)
+    expect(subject).to have_address_changes
```
Comment (munckymagik) [inline spec/support/shared_examples/geocodable_examples.rb]: Let's change this to `have_changed` so we're using the Rails Dirty API directly. `has_address_changes?` could be broken in the model we're testing.

Comment (urfolomeus) [inline spec/support/shared_examples/geocodable_examples.rb]: Good shout.

File: app/models/office.rb (hunk 1 above)
Comment (munckymagik) [inline app/models/office.rb]: Excuse my simplistic approach. What happens if you comment out that line?

Comment (urfolomeus) [inline app/models/office.rb]: It breaks :)

Comment (munckymagik) [inline app/models/office.rb]: Fair enough :)

**Comment by munckymagik** [comment]
:+1:

**Comment by Tr4pSt3R** [comment]
:+1:

**Comment by urfolomeus** [comment]
Thanks @munckymagik @Tr4pSt3R :)

**Pull request merged**